File: 900/2047B/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n;
        cin >> n;
        string s;
        cin >> s;

        vector<int> freq(26, 0);
        for (char c : s)
            freq[c - 'a']++;

        int mini = s.size();
        int maxi = 0;
        char min_ch = ' ';
        char max_ch = ' ';

        for (int i = 0; i < 26; i++) 
        {
            if (freq[i] == 0) 
                continue;

            if (freq[i] < mini) 
            {
                mini = freq[i];
                min_ch = char('a' + i);
            }

            if (freq[i] >= maxi) 
            {
                maxi = freq[i];
                max_ch = char('a' + i);
            }
        }

        for (int i = 0; i < n; i++) 
        {
            if (s[i] == min_ch) 
            {
                s[i] = max_ch;
                break;
            }
        }

        cout << s << "\n";
    }

    return 0;
}

File: 900/2047B/main.java
import java.util.*;
public class main
{
	public static void main(String args[])
	{
		Scanner sc = new Scanner(System.in);
		int t = sc.nextInt();
 
		while(t-- > 0)
		{
			int n = sc.nextInt();
			String s = sc.next();
 
			int freq[] = new int[26];
			for(int i=0;i<s.length();i++)
				freq[s.charAt(i)-'a']++;
 
			int mini = s.length();
			int maxi = 0;
 
			char min_ch = ' ';
			char max_ch = ' ';
 
			for(int i=0;i<26;i++)
			{
				if(freq[i] == 0)   
				    continue;
 
				if(freq[i] < mini)
				{
					mini = freq[i];
					min_ch = (char)('a'+i);
				}
 
				if(freq[i] >= maxi)
				{
					maxi = freq[i];
					max_ch = (char)('a'+i);
				}
			}
 
			StringBuilder sb = new StringBuilder(s);
			for(int i=0;i<sb.length();i++)
			{
				if(sb.charAt(i) == min_ch)
				{
					sb.setCharAt(i,max_ch);
					break;
				}
			}
 
			System.out.println(sb.toString());
		}
	}
}